subprocess.cc

```diff
@@ -263,14 +263,16 @@ void process_run_subcommand(Context<E> &ctx, int argc, char **argv) {
   // Get the mold-wrapper.so path
   std::string self = get_self_path(ctx);
   std::string dso_path;
+  struct stat st;
   if (self == "/usr/bin/mold")
     dso_path = "/usr/lib/mold/mold-wrapper.so";
   else if (self == "/usr/local/bin/mold")
     dso_path = "/usr/local/lib/mold/mold-wrapper.so";
   else
-    dso_path = std::string(path_dirname(self)) + "/mold-wrapper.so";
```

Comment:
This is for the condition that `mold` is in the same directory as `mold-wrapper.so`, just like after you built mold but before install it. So you can't remove this logic. Instead, please remove the special logic for `/usr/bin/mold` and `/usr/local/bin/mold` and instead add a check for the existence of `path_clean(self + "/../../../mold-wrapper.so")`.

Reply:
Sorry, I mean `path_clean(self + "/../../lib/mold/mold-wrapper.so")`